src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -52,10 +56,46 @@ class CosmasGoogleCloudService(private val bucketName: String,
         responseObserver.onCompleted()
     }
 
+    override fun createPatch(request: CosmasProto.CreatePatchRequest,
+                             responseObserver: StreamObserver<CosmasProto.CreatePatchResponse>) {
+        LOG.info("Get request for create new patch of file # ${request.fileId} by user ${request.userId}")
+        synchronized(this.patchBuffer) {
+            val project = this.patchBuffer.getValue(request.projectId)
+            val patchesList = project[request.fileId] ?: mutableListOf()
+            patchesList.add(CosmasInMemoryService.Patch(request.userId, request.text, request.timeStamp))
+            project[request.fileId] = patchesList
+            this.patchBuffer[request.projectId] = project
+        }
+        val response: CosmasProto.CreatePatchResponse = CosmasProto.CreatePatchResponse
+                .newBuilder()
+                .build()
+        responseObserver.onNext(response)
+        responseObserver.onCompleted()
+    }
+
     override fun commitVersion(request: CosmasProto.CommitVersionRequest, responseObserver: StreamObserver<CosmasProto.CommitVersionResponse>) {
         LOG.info("Get request for commit last version of files in project # ${request.projectId}")
         val project = this.fileBuffer[request.projectId]
+        val patchProject = this.patchBuffer[request.projectId]
 
+        if (patchProject != null) {
+            try {
+                for ((fileId, patchList) in patchProject) {
+                    val outputStream = ByteArrayOutputStream()
+                    val output = ObjectOutputStream(outputStream)
+                    for (patch in patchList) {
+                        output.writeObject(patch)
+                    }
+                    this.storage.create(
```

Comment:
Come on. You write messages with different meaning (patch list vs file version) to the same object. This is a wrong way.